src/ripple/protocol/DPAuth.h

```diff
@@ -0,0 +1,66 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2021 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/protocol/LedgerEntryWrapper.h>
+#include <ripple/protocol/STAccount.h>
+
+namespace ripple {
+
+// Wrapper class for DepositPreAuth Ledger Object. This class encapsulates the std::shared_ptr<SLE> to the actual ledger object.
+template <bool Writable>
+class DPAuthImpl final : public LedgerEntryWrapper<Writable>
```

Comment:
I'm not really excited about `DPAuthImpl` as a name.  If I'm not already thinking about deposit preauthorization the `DP` prefix really doesn't get me oriented.  But `DepositPreauth` is already taken.  `DepositPreAuthorization` is getting kind'a long winded, even for me.

What would you think of `DepPreAuthImpl`?  Still not great.  We could also leave this as `DPAuthImpl` for now and see what folks think during the code review.  Your call.  I guess the good news is that this will usually be a return type declared with `auto`, so it won't show up in the source code a whole lot.

Reply:
Agreed, I couldn't think of a better name, so I rushed through with the first one that popped into my head.
I like `DepPreAuthImpl` too, I'm open to any other suggestions as well 👍